npm/ng-packs/packages/core/src/lib/services/replaceable-components.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, NgZone } from '@angular/core';
+import { Router } from '@angular/router';
+import { ReplaceableComponents } from '../models/replaceable-components';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { noop } from '../utils/common-utils';
+import { map, filter } from 'rxjs/operators';
+import { InternalStore } from '../utils/internal-store-utils';
+
+@Injectable({ providedIn: 'root' })
+export class ReplaceableComponentsService {
+  private state: InternalStore<ReplaceableComponents.ReplaceableComponent[]>;
```

Comment:
Store would be better naming

```suggestion
  private store: InternalStore<ReplaceableComponents.ReplaceableComponent[]>;
```